be/src/exec/pipeline/exchange/sink_buffer.cpp

```diff
@@ -0,0 +1,215 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "exec/pipeline/exchange/sink_buffer.h"
+
+namespace starrocks::pipeline {
+
+SinkBuffer::SinkBuffer(RuntimeState* state, const std::vector<TPlanFragmentDestination>& destinations,
+                       size_t num_sinkers)
+        : _mem_tracker(state->instance_mem_tracker()),
+          _brpc_timeout_ms(std::min(3600, state->query_options().query_timeout) * 1000),
+          _num_uncancelled_sinkers(num_sinkers) {
+    for (const auto& dest : destinations) {
+        const auto& instance_id = dest.fragment_instance_id;
+
+        auto it = _num_sinkers.find(instance_id.lo);
+        if (it != _num_sinkers.end()) {
+            it->second += num_sinkers;
+        } else {
+            _num_sinkers[instance_id.lo] = num_sinkers;
+
+            _request_sequences[instance_id.lo] = 0;
+            _max_processed_sequences[instance_id.lo] = -1;
```

Comment:
add comment to explain -1's significance

Reply:
ok.